### PR TITLE
Update Testframework to handle username password for non-multi factor auth tenants

### PR DESCRIPTION
**Pull request by shahabhijeet**
Enable login to non-multifactor auth tenant using username and password.
Adding tests

### CODE REVIEW

File: src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/TestEnvironment.cs [hunk 1]
```diff
@@ -234,15 +233,18 @@ private void Login()
...
-            Currently we prioritize login as below:
-            1) UserName / Password combination
-            2) ServicePrincipal/ServicePrincipal Secret Key
-            3) Interactive Login (where user will be presented with prompt to login)
-           */
-            #region Login
-            ActiveDirectoryServiceSettings aadServiceSettings = new ActiveDirectoryServiceSettings()
+            //We use this because when login silently using userTokenProvider, we need to provide a well known ClientId for an app that has delegating permissions.
+            //All first party app have that permissions, so we use PowerShell app ClientId
+            string PowerShellClientId = "1950a258-227b-4e31-a9cf-717495945fc2";
```
Comment (brjohnstmsft) [inline src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/TestEnvironment.cs]: @shahabhijeet I'm not sure if this is sufficient to solve the problem, at least for the Search tests. Before these changes, this was the client ID being used, but the Tenant ID was blank. I don't think this client ID works with the new default Tenant ID, at least for the account we use for Search tests.

Comment (markcowl) [inline src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/TestEnvironment.cs]: So, if you are using a tenant other than the microsoft tenant, you will need to provide the tenant id

Comment (markcowl) [inline src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/TestEnvironment.cs]: The powershell client id exists in all tenants, it is a first party application

File: src/TestFramework/TestFramework.Tests/TestEnvironment/TestEnvironmentTests.cs [hunk 2]
```diff
@@ -28,5 +28,17 @@ public void DefaultTenantInTestEnvironment()
             Assert.Equal<string>("72f988bf-86f1-41af-91ab-2d7cd011db47", tenantId);
         }
 
+        [Fact(Skip = "environmentsetting string needs to be set from credentials from keyvault for domain that still have userName/Password and 2factor Auth is disabled")]
+
```
Comment (markcowl) [inline src/TestFramework/TestFramework.Tests/TestEnvironment/TestEnvironmentTests.cs]: remove extra line

File: src/TestFramework/TestFramework.Tests/TestEnvironment/TestEnvironmentTests.cs [hunk 3]
```diff
@@ -28,5 +28,17 @@ public void DefaultTenantInTestEnvironment()
...
+        [Fact(Skip = "environmentsetting string needs to be set from credentials from keyvault for domain that still have userName/Password and 2factor Auth is disabled")]
+        
+        public void LoginUsnPwd()
+        {
+            //This environmentsetting string needs to be set from credentials from keyvault for domain that still have userName/Password and 2factor Auth is disabled        
+            HttpMockServer.Mode = HttpRecorderMode.Record;
+            TestEnvironment env = TestEnvironmentFactory.GetTestEnvironment();
+            string userName = env.ConnectionString.KeyValuePairs[ConnectionStringKeys.UserIdKey];
+            Assert.False(string.IsNullOrEmpty(userName));
+            Assert.Equal<string>("admin@AzureSDKTeam.onmicrosoft.com", userName);
```
Comment (markcowl) [inline src/TestFramework/TestFramework.Tests/TestEnvironment/TestEnvironmentTests.cs]: remove this extra assertion

File: src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/TestEnvironment.cs (hunk 1 above)
Comment (brjohnstmsft) [inline src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/TestEnvironment.cs]: @markcowl Is that true in Dogfood as well?

In any case, once this change is ready I'll try it and I'll let you know if it works for me.

**Comment by markcowl** [comment]
A minor comment, then LGTM

**Review by markcowl — CHANGES_REQUESTED** [review]

**Pull request merged**